Share code online With Oscript Editor.

genievot opened the pull request:
### **Added a sharing code button to make sharing online easy for users.**

Users can share the code from Oscript editor to directly show it's working to anyone. The code-sharing is powered with free API http://myjson.com which will take the whole agent object and create a router URL for it which can be copied and given to anybody. This will open the website with prewritten AA in it.
![2019-08-04 (14)](https://user-images.githubusercontent.com/40867747/62422118-fb766480-b6ca-11e9-86e4-5c4da49da3be.png)

Because of History mode in Vue, I am using one of the recommended [Firebase hosting](https://cli.vuejs.org/guide/deployment.html#firebase) to tackle the router problem...
**The demo is here to test out:-** https://oscript-editor.firebaseapp.com/


I would like to request that I am not that good with writing the code with best practices, so please adjust it accordingly if you want to or if you think that there is a need for it.
Thank you.

tonyofbyteball commented:
This is a nice feature!  Could you additionally copy the URL to clipboard and briefly display a message like "Copied link to clipboard".  Currently, there is no visual feedback after one clicks the share button and it's not quite evident that the URL changed.  Otherwise, great work!

genievot commented:
> This is a nice feature! Could you additionally copy the URL to clipboard and briefly display a message like "Copied link to clipboard". Currently, there is no visual feedback after one clicks the share button and it's not quite evident that the URL changed. Otherwise, great work!

Sure great thought, I will do it as soon as possible and add a new commit. Thank you.

genievot commented:
Done @tonyofbyteball  please take a look and let me know if something to change. I used simple alert instead of any new component to showing a notification to the user.

genievot commented:
Fixed the uncommon behavior of adding duplicate labels for the same text.

genievot commented:
Added serial number incrementation in this commit which means if code label is already there and someone opens a shared link with same label name then it will be like this...

![2019-08-10 (45)](https://user-images.githubusercontent.com/40867747/62820402-77682500-bb81-11e9-8c7c-a843ea920045.png)


and no wired id will be showing to make it unique.

tonyofbyteball commented on src/store/modules/lib/agents.js:
```diff
@@ -57,11 +60,91 @@ export default () => ({
+					findOldAgentText.text = text
+					findOldAgentText.id = id
+					// This below condition checks. If there is already named label and the old text and if that text === label's state text
+				} else if (findOldAgentLabel && findOldAgentText && findOldAgentLabel.text === findOldAgentText.text) {
```
this in never true after `if (findOldAgentText)` above

genievot replied:
yes, good catch should have followed this.

tonyofbyteball commented on src/store/modules/lib/agents.js:
```diff
@@ -57,11 +60,91 @@ export default () => ({
+					findOldAgentLabel.id = id
+				} else if (findOldAgentLabel) {
+					var inBrackets = findOldAgentLabel.label.trim().match(/\(.*?\)/g)
+					if (findOldAgentLabel || inBrackets) {
```
this is always true due to the condition 2 lines above

tonyofbyteball commented on src/store/modules/lib/agents.js:
```diff
@@ -57,11 +60,91 @@ export default () => ({
+					var inBrackets = findOldAgentLabel.label.trim().match(/\(.*?\)/g)
+					if (findOldAgentLabel || inBrackets) {
+						if (inBrackets) {
+							var wasLabel = findOldAgentLabel.label
```
consider using the existing function getIncrementedLabel instead

genievot replied:
The existing function cannot be used right now. But in future with time, I will try to manage it.

tonyofbyteball commented on index.js:
```diff
@@ -0,0 +1,15 @@
+const electron = require('electron')
```
please keep this PR focused on one subject, move electron to a separate PR

genievot replied:
oops didn't mean to add it, Don't know how it ended up there. Will remove it and fix remaining reviews.

genievot commented:
Fixed the code problems and removed index.js for electron. But the use of current function `getIncrementedLabel` Cannot be used right now. The method checks for integer inside brackets but checks for postfix integer.
